control/velocity_controller/include/velocity_controller/velocity_controller.h

```diff
@@ -181,16 +177,14 @@ class VelocityController
   Lpf1d lpf_vel_error_;
 
   // methods
-  void callbackCurrentVelocity(const geometry_msgs::TwistStamped::ConstPtr & msg);
-  void callbackTrajectory(const autoware_planning_msgs::TrajectoryConstPtr & msg);
-  void callbackTimerControl(const ros::TimerEvent & event);
-  void callbackConfig(
-    const velocity_controller::VelocityControllerConfig & config, const uint32_t level);
+  void callbackCurrentVelocity(const geometry_msgs::msg::TwistStamped::ConstSharedPtr msg);
```

Comment:
I always had to change to `SharedPtr`. I wonder if/why that wouldn't be needed here. Are the callbacks not used?

Reply:
I think we can use `ConstSharedPtr` here. There is also a subscriber using this callback here.
Which is the code you had to change to the `SharedPtr`?

Reply:
I think it was [here](https://github.com/tier4/Pilot.Auto/pull/7/files#diff-357aaf098422617a99a3b414b76d9b642d43edf3699d7cf4e2b67cbf0d2f0300R40) but now I see that it was a `const *`, not `ConstSharedPtr`. Never mind, keep your code as is